### PR TITLE
enable msvc static runtime build for boost

### PR DESCRIPTION
**Pull request by Dargun**
enable the option to build msvc boost libs with static runtime.

### CODE REVIEW

File: cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in [hunk 1]
```diff
@@ -202,6 +202,9 @@ endif()
 
 if(BOOST_BUILD_STATIC)
   set(link_opts link=static)
+  if(NOT BOOST_BUILD_DYNAMIC_VSRUNTIME)
```
Comment (ruslo) [inline cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in]: Do we need to protect it with `if(MSVC)`?

File: cmake/projects/Boost/hunter.cmake [hunk 2]
```diff
@@ -278,6 +279,15 @@ hunter_add_version(
...
+    hunter_cmake_args(
+      Boost
+      CMAKE_ARGS
+        BOOST_BUILD_DYNAMIC_VSRUNTIME=${_hunter_vs_md}
+    )
+endif()
+
 hunter_pick_scheme(DEFAULT url_sha1_boost)
 hunter_cacheable(Boost)
 hunter_download(PACKAGE_NAME Boost PACKAGE_INTERNAL_DEPS_ID "19")
```
Comment (ruslo) [inline cmake/projects/Boost/hunter.cmake]: Please update `PACKAGE_INTERNAL_DEPS_ID`

File: cmake/projects/Boost/hunter.cmake [hunk 3]
```diff
@@ -278,6 +279,15 @@ hunter_add_version(
     26a52840e9d12f829e3008589abf0a925ce88524
 )
 
+if(MSVC)
+    hunter_check_toolchain_definition(NAME "_DLL" DEFINED _hunter_vs_md)
```
Comment (ruslo) [inline cmake/projects/Boost/hunter.cmake]: Indentation 2 here:
* https://0.readthedocs.io/en/latest/cmake.html#indentation

File: cmake/projects/Boost/hunter.cmake [hunk 4]
```diff
@@ -278,6 +279,15 @@ hunter_add_version(
...
+    hunter_cmake_args(
+      Boost
```
Comment (ruslo) [inline cmake/projects/Boost/hunter.cmake]: Indentation 4 here.

File: cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in (hunk 1 above)
Comment (Dargun) [reply cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in]: no not realy, but it's an visual studio feature only. the check is not necessary.

Comment (ruslo) [reply cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in]: ok

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2877
* https://travis-ci.org/ingenue/hunter/builds/307514673?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2878
* https://travis-ci.org/ingenue/hunter/builds/307612239?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.171

**Comment by s-brain** [comment]
This has broken my builds, with no changes other than updating hunter i now get static-runtime builds, 
I can see it is looking for a "_DLL" item in the toolchain, how do i make sure this is set by default.

**Comment by ruslo** [comment]
> This has broken my builds

Tests looks good, if you think that there is an error please show minimal example to reproduce.

> I can see it is looking for a "_DLL" item in the toolchain

To be precise for the definition in Toolchain-ID.

> how do i make sure this is set by default

`_DLL` defined if `/MD` flag used, this is default. If you want to build with `/MT` use toolchain, example:
* https://github.com/ruslo/polly/blob/master/vs-12-2013-mt.cmake

**Comment by s-brain** [comment]
Ok My mistake, I want /MD, but something had added /MT to my toolchain, but it was being overriden later.
Cheers

**Pull request merged**